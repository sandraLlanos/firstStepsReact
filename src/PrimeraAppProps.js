import React from 'react';

/**Propiedaddes 
 * son las propiedades que son enviadas a los componentes
 */

//  const PrimeraAppProps = ( props ) => { usando props directamente
// const PrimeraAppProps = ( {saludo } ) => { usando desestructuracion
const PrimeraAppProps = ( {saludo = 'hola mundo'} ) => { //pasando un valor por defecto

   // const saludo = '!Hello World';
   // console.log( props );
   console.log( saludo );
   
    return (
       <>
          {/* <h1>{ props.saludo }</h1> */}
          <h1>{ saludo }</h1>
          
       </>
    );
 }
 export default PrimeraAppProps;