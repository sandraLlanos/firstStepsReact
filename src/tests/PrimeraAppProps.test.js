import React from 'react';
import { shallow } from 'enzyme';
import PrimeraAppProps from '../PrimeraAppProps';

describe('pruebas sobre <PrimeraAppProps />', () => {
    // test('debe de mostrar el mensaje "hola soy Sandra" ', () => {
        
    //     const saludo = 'Hola soy Sandra';
    //     // wrapper en este caso seria el prodicto renderizado de lo que se desea mostrar,
    //     // es decir del compente de react.
    //     // para esto es neceserio importar render de @testing-library/react
    //     // render recibe el elemento que se  desea renderizar
    //     // const wrapper = render( <PrimeraAppProps saludo={saludo}/>);
    //     const { getByText } = render( <PrimeraAppProps saludo={saludo}/>);

    //     // para esto se requiere la importacion de import '@testing-librar/jest-dom/extend-expect' de setupTest.js
    //     // expect( getByText(saludo) ).toBeInTheDocument();
    // });

    test('debe de mostrar <PrimeraAppProps /> correctamete ', () => {

        const saludo = 'Hola soy Sandra';
        const wrapper =  shallow( <PrimeraAppProps saludo={saludo}/> );

        expect( wrapper ).toMatchSnapshot();
    })

    
    
})
