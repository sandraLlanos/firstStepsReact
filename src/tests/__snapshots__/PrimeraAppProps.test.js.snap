// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre <PrimeraAppProps /> debe de mostrar <PrimeraAppProps /> correctamete  1`] = `
<Fragment>
  <h1>
    Hola soy Sandra
  </h1>
  <p>
    Mi primer aplicaci√≥n
  </p>
</Fragment>
`;
